specs/badge.md

```diff
@@ -0,0 +1,80 @@
+# fast-badge
+
+## Overview
+
+*Badge* component is used to highlight an item and attract attention or flag status, such as “New”, “Sale”, or other short phrases.
+
+### Use Cases
+Keep the *badge* label short and succinct &ndash; it should not be a sentence or a long phrase (recommend 2-3 words). Short phrases work well, such as “App of the day,” but keep in mind that they will generally become longer when localized. Keep the text to nouns and adjectives if possible (e.g. “New” or “New today”), so the user won’t think it is an actionable element.
+
+Typical use cases include, but are not limited to, denoting a sale or new item and flagging an item as part of a category
+  
+### Features
+
+A badge should allow the following properties:
+- `fill`, a string that passes the color value for the backplate. Defaults to "transparent".
+- `size`, the size the badge is rendered.
+
+### Risks and Challenges
+
+It's import to not over dominate the users screen with the use of *badge*. They should be used sparingly to denote special status or draw attention to a particular item.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/badge)
+- [Material UI](https://material-ui.com/components/badges/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/badges/)
+- [Ant Design Badge](https://ant.design/components/badge/)
+- [Ant Design Lozenge](https://atlaskit.atlassian.com/packages/core/lozenge)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/badge)
+
+---
+
+### API
+
+*Component name:*
+- `fast-badge`
+
+*Attributes:*
+- `fill`: string
```

Comment:
Should this be handled with a CSS custom property rather than an HTML attribute? If handled with a CSS prop then the consumer can style these like any other. Not sure how we would handle size that way, if the values are a string enum. Fill should be easy enough. If we don't use a CSS prop here, I think we should have a strong justification for that.

Reply:
I was chatting offline with @eljefe223  about this yesterday. I'd agree - I don't think a "fill" property that accepts a color string provides much utility. As you say, you could just as easily style it:
```css
fast-badge { --fill: red; }
/* or */
fast-badge { background-color: red }
```
or
```html
<fast-badge style="background: red;" />
```

The struggle we've always had around badge is how to make common app scenarios easy to apply but also make it intuitively extensible. One idea is to have a "fill" property map to author-defined custom properties:

```css
fast-badge {
   --fill-foo: red;
   --fill-bar: green;
   --fill-transparent: transparent
}
```

Then in HTML, those could easily used:
```html
<fast-badge fill="transparent" />
<fast-badge fill="foo" />
```

You could still use any of the above css override techniques to create custom instances.

Reply:
Opening this back up because I think it represents the same question about how we support different appearances on any component. We've talked about "style sets" or "appearances", essentially a grouping of styling properties. It's not as simple as just setting a fill because the foreground needs to respond. Also, if setting a fill, it's probable that the value is used elsewhere.

The closest parallel I think would be a good example is how we describe an input to either be filled or outlined. Is that an "appearance" property, or is it just in css? Do we encourage people to make two templates and derive components like FilledInput and OutlinedInput?

Or perhaps eventually it's more nuanced than that and the appearance might be determined by the context, like on L1 it's outlined and on L2 and above it's filled. Obviously very specific example, but the general idea being that some logic in the component or styling can determine the default set. Of course someone could override and customize as necessary with css, etc.